pytorch_lightning/callbacks/pt_callbacks.py

```diff
@@ -254,6 +255,30 @@ def on_epoch_end(self, epoch, logs=None):
                 self.save_model(filepath, overwrite=False)
 
 
+class GradientAccumulationScheduler(Callback):
+    """Change gradient accumulation factor according to scheduling.
+    # Arguments
+        scheduling: dict, scheduling in format {epoch: accumulation_factor}
+    """
+    def __init__(self, scheduling: dict):
+        minimal_epoch = min(scheduling.keys())
+        if minimal_epoch < 1:
```

Comment:
I would understand that it should not be negative, but it can be 0, right?
it would be nice to have a check that the keys are integers....

Reply:
Thank you for rewiew.
Add TypeError, if scheduling epochs or accumulation factor isnt integer